src/natcap/invest/urban_nature_access.py

```diff
@@ -165,29 +189,289 @@ def execute(args):
         n_workers = -1  # Synchronous execution
     graph = taskgraph.TaskGraph(work_token_dir, n_workers)
 
+    kernel_types = {
+        'dichotomy': instantaneous_decay_kernel_raster,
+        # "exponential" is more consistent with other InVEST models'
+        # terminology.  "Power function" is used in the design doc.
+        'exponential': utils.exponential_decay_kernel_raster,
+        'gaussian': utils.gaussian_decay_kernel_raster,
+        'density': density_decay_kernel_raster,
+    }
+    # Since we have these keys defined in two places, I want to be super sure
+    # that the labels match.
+    assert sorted(kernel_types.keys()) == (
+        sorted(ARGS_SPEC['args']['decay_function']['options']))
+
+    if 'kernel_type' not in args:
+        kernel_type = 'dichotomy'
+    elif args['kernel_type'] not in kernel_types:
+        raise ValueError(
+            f'Kernel type "{args["kernel_type"]}" is not recognized. '
+            f"Must be one of {', '.join(kernel_types.keys())}")
+    else:
+        kernel_type = args['kernel_type']
+
     # Align the population raster to the LULC.
     lulc_raster_info = pygeoprocessing.get_raster_info(
         args['lulc_raster_path'])
 
+    squared_lulc_pixel_size = _square_off_pixels(args['lulc_raster_path'])
+    lulc_alignment_task = graph.add_task(
```

Comment:
I was just curious if we'll ever want to make sure that the LULC and population raster align perfectly geospatially by using `align_and_resize_rasters`?

Reply:
Yeah, this has been on my mind as well.  In the interest of closing out this PR, might a `TODO` suffice?  In case it does, I've added one to 80adcfdf